src/VisualStudio/Core/Def/ProjectSystem/Legacy/AbstractLegacyProject.cs

```diff
@@ -147,7 +147,9 @@ internal abstract partial class AbstractLegacyProject : ForegroundThreadAffiniti
         ConnectHierarchyEvents();
         RefreshBinOutputPath();
 
-        _externalErrorReporter = new ProjectExternalErrorReporter(ProjectSystemProject.Id, externalErrorReportingPrefix, language, workspaceImpl);
+        var projectHierarchyGuid = GetProjectIDGuid(hierarchy);
+
+        _externalErrorReporter = new ProjectExternalErrorReporter(ProjectSystemProject.Id, projectHierarchyGuid, externalErrorReportingPrefix, language, workspaceImpl);
```

Comment:
hierarchy guid is required for the current project filter to work in the error list.